odbc/src/api/data.rs

```diff
@@ -196,9 +198,58 @@ impl IntoCData for Bson {
     fn to_datetime(&self) -> Result<DateTime<Utc>> {
         match self {
             Bson::DateTime(d) => Ok((*d).into()),
-            Bson::String(s) => Utc
-                .datetime_from_str(s, "%FT%H:%M:%S%.3fZ")
-                .map_err(|_| ODBCError::InvalidDatetimeFormat(s.clone())),
+            Bson::String(s) => {
+                if s.contains('T') {
+                    // received a datetime string
+                    Utc.datetime_from_str(s, "%FT%H:%M:%S%.3fZ")
+                        .map_err(|_| ODBCError::InvalidDatetimeFormat(s.clone()))
+                } else if s.contains('-') {
+                    // received a date string
+                    let date = NaiveDate::parse_from_str(s, "%F")
+                        .map_err(|_| ODBCError::InvalidDatetimeFormat(s.clone()))?;
+                    Ok(DateTime::<Utc>::from_utc(date.and_hms(0, 0, 0), Utc))
+                } else {
+                    // received a time string -- should return utc today + time
+                    let time = NaiveTime::parse_from_str(s, "%H:%M:%S%.3f")
+                        .map_err(|_| ODBCError::InvalidDatetimeFormat(s.clone()))?;
+                    Ok(DateTime::<Utc>::from_utc(
+                        NaiveDateTime::new(Utc::today().naive_utc(), time),
+                        Utc,
+                    ))
+                }
+            }
+            o => Err(ODBCError::RestrictedDataType(o.to_type_str(), DATETIME)),
+        }
+    }
+
+    fn to_date(&self) -> Result<NaiveDate> {
+        match self {
+            Bson::DateTime(d) => Ok((*d).to_chrono().date_naive()),
```

Comment:
From what I can see in the [spec](https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/sql-to-c-timestamp?view=sql-server-ver15), if the time portion of the timestamp is nonzero, then it should return SUCCESS_WITH_INFO - Truncated data (01S07).